homeassistant/components/emoncms/sensor.py

```diff
@@ -163,10 +163,7 @@ def __init__(
         self._sensorid = sensorid
         self._elem = elem
 
-        if unit_of_measurement == "kWh":
-            self._attr_device_class = SensorDeviceClass.ENERGY
-            self._attr_state_class = SensorStateClass.TOTAL_INCREASING
-        elif unit_of_measurement == "Wh":
+        if unit_of_measurement == "kWh" or unit_of_measurement == "Wh":
```

Comment:
```suggestion
        if unit_of_measurement in ("kWh", "Wh"):
```

The same for temperature below